test-functional.sh

```diff
@@ -10,7 +10,7 @@ do
     TEST_CMD=(python -m pytest -v --random-order-bucket global "$f")
     echo "${TEST_CMD[@]}"
     if command -v xvfb-run > /dev/null; then
-        xvfb-run -a "${TEST_CMD[@]}"
+        xvfb-run  --server-args="-screen 0, 1680x1050x24" -a "${TEST_CMD[@]}"
```

Comment:
note: `xvfb-run` is Linux only, meaning that we'd want to xfail or skip any tests that require a specific screen resolution (or investigate another approach for on mac)

Reply:
with the latest changes, this isn't strictly necessary (tests should pass on common screen sizes), but we can leave it in